ssz_serialization/merkleization.nim

```diff
@@ -176,11 +176,10 @@ func addChunk*(merkleizer: var SszMerkleizerImpl, data: openArray[byte]) =
         merkleizer.combinedChunks[i] = hash
         break
   else:
-    let paddingBytes = bytesPerChunk - data.len
+    assign(merkleizer.combinedChunks[0].data.toOpenArray(0, data.high), data)
 
-    merkleizer.combinedChunks[0].data[0..<data.len] = data
-    merkleizer.combinedChunks[0].data[data.len..<bytesPerChunk] =
-      zero64.toOpenArray(0, paddingBytes - 1)
+    for i in data.len..<bytesPerChunk:
```

Comment:
isn't the `assign` trick applicable here as well?

Reply:
it doesn't really make a difference, the vast majority of data is 32 bytes exactly and doesn't enter this loop

Reply:
also, this check is optimized away because the target length is known at compile time, so the generated code for setting to 0 is ok